run.py

```diff
@@ -200,7 +200,10 @@ def add_optimizer(self, parameters):
 		Returns an optimizer for learning the parameters of the model
 		
 		"""
-		return torch.optim.Adam(parameters, lr=self.p.lr, weight_decay=self.p.l2)
+		if self.p.optimizer == 'Adagrad':
+			return torch.optim.Adagrad(parameters, lr=self.p.lr, weight_decay=self.p.l2)
+		else:
```

Comment:
Could we use an else if ... == 'Adam' here as well instead of else? If someone makes a spelling error in Adagrad, this would lead to the script using Adam without the user noticing.

Reply:
Added this, also added an error out in case the optimizer param is NOT Adam or Adagrad